<launch>
  <arg name="use_sim" default="true" />

  <node name="active_perception" pkg="robot_manipulation" type="active_perception_service" output="screen">
    <remap from="/debug_cloud" to="/active_perception/debug_cloud" />
    <remap from="/color_image" to="/camera1/color/image_raw"/>
    <remap from="/depth_image" to="/camera1/aligned_depth_to_color/image_raw"/>
    <remap from="/camera_info" to="/camera1/color/camera_info"/>

    <param name="crop_min_x" type="double" value="0.0" />
    <param name="crop_min_y" type="double" value="-0.17" />
    <param name="crop_min_z" type="double" value="0.0" />
    <param name="crop_max_x" type="double" value="0.5" />
    <param name="crop_max_y" type="double" value="0.19" />
    <param name="crop_max_z" type="double" value="0.15" />
    <param name="voxel_size" type="double" value="0.003" />
    <param name="base_link_frame" type="str" value="wx250s/base_link" />
  </node>

  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_detector" output="screen">
    <remap from="image_rect" to="/camera1/color/image_raw"/>
    <remap from="camera_info" to="/camera1/color/camera_info"/>

    <param name="tag_family" type="str" value="tag16h5" />
    <param name="camera_frame" type="str" value="camera1" />
    <param name="publish_tf" type="bool" value="false" />
    <param name="standalone_tags" type="yaml" value="[{id: 29, size: 0.075}]" />
    <param name="publish_tag_detections_image" type="bool" value="true" />
    <param name="tag_refine_pose" type="int" value="1" />
    <param name="max_hamming_dist" type="int" value="5" />
  </node>

  <node name="object_reg_node" pkg="robot_manipulation" type="object_reg" output="screen">
    <param name="mesh_folder" type="str" value="$(find robot_manipulation)/resources/meshes" />

    <remap from="/point_cloud" to="/active_perception/debug_cloud" />
    <remap from="/rviz_debug" to="/object_reg/rviz_debug" />
    <remap from="/filtered_point_cloud" to="/object_reg/filtered" />
    <remap from="/detections" to="/object_reg/detections" />
  </node>

  <node name="grasp_gen_service" pkg="robot_manipulation" type="grasp_gen" output="screen">
    <param name="config_file" type="str" value="$(find robot_manipulation)/resources/ros_grasp.cfg" />
  </node>

  <!-- Launch the base functionality -->
  <include file="$(find robot_manipulation)/launch/base.launch">
    <arg name="use_rviz" value="true" />
    <arg name="use_sim" value="$(arg use_sim)" />
    <arg name="dual_camera" value="true" />
  </include>
</launch>
